### PR TITLE
Show which beatmap packs have been cleared & update "clear" badge design

### LINKED ISSUE #2548: "Completed/All-Played" Beatmap Packs status
Here is a QOL improvement I want to suggest.

I want to have Beatmap Packs indicate if I have completed all maps in that Pack, and if I do, it will show a "Completed" or "All-Played" status, like this in the picture below.

capture

### LINKED ISSUE #2758: Cleared/Uncleared maps in Beatmap Packs can be hard to tell apart
image

The shades of gray and green are very similar, the cleared ones do not stand out very much when looking at a list of maps in a pack as a whole.
It's difficult to skim over when trying to quickly judge what maps need to be played to complete a pack.

### PR DESCRIPTION
**Pull request by cl8n**
closes #2548, fixes #2758
17ec74
11023d

the clear badges keep their width between cleared/uncleared by making the text invisible... not sure if there was a better way to do that

for logged out users all of the empty badges and gray stripes look strange I think I'll hide them

### CODE REVIEW

**Comment by peppy** [comment]
@arflyte please check this one design-wise

**Comment by cl8n** [comment]
each pack has to make that `withHasCompleted` select query twice now, once when loading the page and again when showing the pack's contents. should I just load all of the content in the first request, or is there a reason it wasn't done like that originally?

File: resources/assets/less/bem/beatmap-pack-items.less [hunk 1]
```diff
@@ -22,15 +22,34 @@
...
+  }
+
+  &__status {
+    background-color: @osu-colour-b5;
+    border-radius: 10000px;
+    display: inline-block;
+    padding: 0 0.5em;
+  }
+
+  &__set--cleared &__status {
```
Comment (nanaya) [inline resources/assets/less/bem/beatmap-pack-items.less]: should be defined inside `__status`.

File: resources/assets/less/bem/beatmap-pack.less [hunk 2]
```diff
@@ -51,11 +52,24 @@
     font-size: 90%;
   }
 
+  &__clear {
```
Comment (nanaya) [inline resources/assets/less/bem/beatmap-pack.less]: Make it modifier of `__cell`. Also for others which for some reason are two elements (of same block) at same time.

**Comment by nekodex** [comment]
btw, I think you may have misunderstood flyte's design - the relevant mode icon should still be shown (left of clear), just not the _other_ unrelated modes.

**Comment by notbakaneko** [comment]
Closing due to lack of updates

**Pull request closed without merging**